forem/forem
# Fix reply button visibility on locked discussion

## What type of PR is this? (check all applicable)


## Description

This PR fixes the issue that reply buttons are still visible on comment threads for a locked discussion. They should not appear when a discussion is locked. 

This PR also hides the new comment box for locked discussions on legacy comment threads.

## Related Tickets & Documents



- Closes #17891

## QA Instructions, Screenshots, Recordings
1. Navigate to an article that has its discussion locked.
2. Navigate to any comment thread. 
3. Observe that reply buttons are not present. Example screenshot below:
![screenshot 1](https://user-images.githubusercontent.com/88938117/176815470-035f0c76-227d-44c8-917d-6b493c44ffac.jpg)


4. On legacy comment threads, observe also that the new comment box is not shown. Example screenshot below:
![screenshot 2](https://user-images.githubusercontent.com/88938117/176815496-1d739709-bc8d-4129-8047-26f8c31cdfda.jpg)


Tested on Chrome and Firefox.

### UI accessibility concerns?

None I am aware of.

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [[Developer Docs](https://developers.forem.com/)](https://developers.forem.com) or
      [[Storybook](https://storybook.forem.com/)](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [[Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`[CHANGELOG.md](https://github.com/forem/forem/tree/main/CHANGELOG.md)`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [[Changelog](https://forem.dev/t/changelog)](https://forem.dev/t/changelog)
      or in a [[forem.dev](http://forem.dev/)](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?
